app/src/main/java/io/github/droidkaigi/confsched2018/presentation/MainViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package io.github.droidkaigi.confsched2018.presentation
+
+import android.annotation.SuppressLint
+import android.arch.lifecycle.Lifecycle
+import android.arch.lifecycle.LifecycleObserver
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.OnLifecycleEvent
+import android.arch.lifecycle.ViewModel
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.SharedPreferences
+import io.github.droidkaigi.confsched2018.R
+import io.github.droidkaigi.confsched2018.presentation.common.pref.Prefs
+import javax.inject.Inject
+
+@SuppressLint("StaticFieldLeak")
+class MainViewModel @Inject constructor(
+        // this field is application context not activity context
+        private val applicationContext: Context
+) : ViewModel(), LifecycleObserver {
```

Comment:
I didn't check it. But If we can, Please use AndroidViewModel for this purpose 👍

Reply:
I didn't know there was AndroidViewModel class, thank you for telling me @takahirom 👍